src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -665,11 +686,18 @@ private PackageDependency CreateDependencyFromProject(dynamic project, Dictionar
                 projectFactory.ProjectProperties = ProjectProperties;
                 projectFactory.BuildProject();
                 var builder = new PackageBuilder();
-                try
+                if (!Directory.Exists(projectFactory.TargetPath))
```

Comment:
If it's packing an xproj, then it ends up building multiple versions of the dll, one for each target framework.  The code normally points the TargetPath to the built dll, but if there are potentially more than one, it points to the directory where the target framework folders will be.  In that case, it doesn't need to extract the metadata from a dll but will instead rely on the project and json files for metadata.  I can comment this to make it more clear.